libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -120,6 +122,76 @@ void AP_DDS_Client::update_topic(sensor_msgs_msg_NavSatFix& msg, const uint8_t i
     msg.position_covariance[8] = vdopSq;
 }
 
+void AP_DDS_Client::update_topic(sensor_msgs_msg_BatteryState& msg, const uint8_t instance)
+{
+    if (instance >= AP_BATT_MONITOR_MAX_INSTANCES) {
+        return;
+    }
+
+    update_topic(msg.header.stamp);
+    auto &battery = AP::battery();
+    
+    if (!battery.healthy(instance))
+    {
+        msg.power_supply_status = 3; //POWER_SUPPLY_HEALTH_DEAD
+        msg.present = false;
+        return;
+    }
+    msg.present = true;
+
+    msg.voltage = battery.voltage(instance);
+    
+    float temperature; 
+    msg.temperature = (battery.get_temperature(temperature, instance)) ? temperature : NAN;
+    
+    float current;
+    msg.current = (battery.current_amps(current, instance)) ? -1 * current : NAN;
+    
+    float capacity = (float)battery.pack_capacity_mah(instance)/1000.0;
+    msg.capacity = capacity;
+
+    uint8_t percentage;
+    if (battery.capacity_remaining_pct(percentage, instance))
+    {
+        msg.percentage = percentage;
+        msg.charge = (percentage/100.0) * (capacity);
+    }
+    else
+    {
+        msg.percentage = NAN;
+        msg.charge = NAN;
+    }
+    
+    msg.design_capacity = NAN;
+ 
+    if (battery.current_amps(current, instance))
+    {
+        if (percentage == 100) {msg.power_supply_status = 4;} //POWER_SUPPLY_STATUS_FULL
+        else if (-1*current > 0.0) {msg.power_supply_status = 1;} //POWER_SUPPLY_STATUS_CHARGING
+        else if (-1*current < 0.0) {msg.power_supply_status = 2;} //POWER_SUPPLY_STATUS_DISCHARGING
+        else {msg.power_supply_status = 3;} //POWER_SUPPLY_STATUS_NOT_CHARGING
+    }
+    else
+    {
+        msg.power_supply_status = 0; //POWER_SUPPLY_STATUS_UNKNOWN
+    }
+
+    msg.power_supply_health = (battery.overpower_detected(instance)) ? 4 : 1; //POWER_SUPPLY_HEALTH_OVERVOLTAGE or POWER_SUPPLY_HEALTH_GOOD
+
+    msg.power_supply_technology = 0; //POWER_SUPPLY_TECHNOLOGY_UNKNOWN
+
+
+    if (battery.has_cell_voltages(instance))
+    {
+        const uint16_t* intVoltages = battery.get_cell_voltages(instance).cells;
+        float floatVoltages[sizeof(intVoltages)];
+        for (int i=0; i < (int)sizeof(intVoltages); i++) 
```

Comment:
Want to use the array size macro `ARRAY_SIZE`?

Reply:
Looking into AP_BattMonitor.h I found that the cells struct is defined like this:
```
// cell voltages in millivolts
struct cells {
    uint16_t cells[AP_BATT_MONITOR_CELLS_MAX];
};
```
So getting the size of the array is unnecessary, as it is always AP_BATT_MONITOR_CELLS_MAX